sound/soc/bcm/allo-boss-dac.c

```diff
@@ -275,23 +275,40 @@ static int snd_allo_boss_hw_params(
 	struct snd_soc_pcm_runtime *rtd = substream->private_data;
 	int channels = params_channels(params);
 	int width = snd_pcm_format_width(params_format(params));
+	struct snd_soc_component *component = asoc_rtd_to_codec(rtd, 0)->component;
+	struct snd_soc_card *card = rtd->card;
 
-	if (snd_soc_allo_boss_master) {
-		struct snd_soc_component *component = asoc_rtd_to_codec(rtd, 0)->component;
+	// Mute before changing sample rate
+	snd_allo_boss_gpio_mute(card);
 
-		snd_allo_boss_set_sclk(component,
-			params_rate(params));
+	if (snd_soc_allo_boss_master) {
+		snd_allo_boss_set_sclk(component, params_rate(params));
 
-		ret = snd_allo_boss_update_rate_den(
-			substream, params);
-		if (ret)
+		ret = snd_allo_boss_update_rate_den(substream, params);
+		if (ret) {
+			// Unmute if error occurs
```

Comment:
It's perfectly acceptable to use `goto`s for error handling. It would work well in this function to put a label (`error:`, perhaps) before the unmute at the end. Then, all of the:
```
	if (ret) {
		// Unmute if error occurs
		snd_allo_boss_gpio_unmute(card);
		return ret;
	}
```
can become:
```
	if (ret)
		goto error;
```

Reply:
Nice to know @pelwell, I thought goto's weren't liked over here : D

Reply:
See: https://www.kernel.org/doc/html/v6.6/process/coding-style.html#centralized-exiting-of-functions

Reply:
I've improved commenting (following what @popcornmix sent) and added gotos like @pelwell mentioned.